### PR TITLE
feat(auth): add Clerk authentication to [DELETE] /api/v1/git_tokens/{git_label_id}

### PR DESCRIPTION
**Pull request by mohamad-jaafar-monty**
Summary:
- updated API '[DELETE] /api/v1/git_tokens/{git_label_id}' to start using Clerk authentication.

- Exception was being swallowed and not logged, added logging. 

- Fixed Unit Tests related to this API

- Moved the unit test to a proper structure

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved token masking in API responses for better security and consistency.
	- Enhanced error handling and logging for git label operations.
- **Tests**
	- Added comprehensive tests for deleting git labels, covering success, error, and edge cases.
	- Updated test mocks to use dynamic token masking.
	- Removed outdated delete endpoint tests and improved test structure.
- **Chores**
	- Added a reusable test fixture for sample token IDs.

### CODE REVIEW

File: app/routes/git_tokens.py [hunk 1]
```diff
@@ -86,7 +89,9 @@ async def handle_gitlab(
         return APIResponse.success(
             message=constants.TOKEN_SAVED_SUCCESSFULLY, data={"id": str(git_label.id)}
         )
-    except Exception:
+    except Exception as e:
+        logger.exception("Unexpected Failure while attempting to save GitLab token on Path = '[POST] /api/v1/git_tokens' -> handle_gitlab")
+        
         return APIResponse.error(
             message=f"Failed to save GitLab token: {str(e)}",
```
Comment (Hayat55) [inline app/routes/git_tokens.py]: Don't return  raw exception message in the API response instead of this add it to log files

**Review by Hayat55 — CHANGES_REQUESTED** [review]

File: app/routes/git_tokens.py [hunk 2]
```diff
@@ -4,340 +4,354 @@
...
-    # This could be from JWT token, session, or other auth mechanism
-    return "user_2sw6NOnSajM1kpsLPA1ZnxCW3uZ"
+	"""
+	Dependency to get current user ID.
+	In a real application, this would extract user ID from JWT token or session.
+	For now, this is a placeholder that you should implement based on your auth system.
+	"""
+	# Will be changed
+	# This could be from JWT token, session, or other auth mechanism
+	return "user_2sw6NOnSajM1kpsLPA1ZnxCW3uZ"
```
[the remark this answers is not in the document]
Comment (mohamad-jaafar-monty) [reply app/routes/git_tokens.py]: It will be removed in the future, its just a placeholder
